src/System.Runtime.Extensions/src/System/IO/TextReader.cs

```diff
@@ -87,6 +87,16 @@ public virtual int Read(char[] buffer, int index, int count)
                 throw new ArgumentException(SR.Argument_InvalidOffLen);
             }
 
+            return Read(new Span<char>(buffer, index, count));
+        }
+
+        // Reads a span of characters. This method will read up to
+        // count characters from this TextReader into the
+        // buffer character array starting at position
+        // index. Returns the actual number of characters read.
+        //
+        public virtual int Read(Span<char> destination)
```

Comment:
This approach needs to be inverted.  We're adding a virtual method here, which means existing derivations of TextReader may have overridden `Read(char[], ...)` but couldn't yet have overridden `Read(Span<char>)`.  If new code then calls `Read(Span<char>)` instead of `Read(char[], ...)`, it won't get the derived behavior but rather than potentially-incorrect/potentially-inefficient base implementation.  So the base `Read(Span<char>)` needs to delegate to the base `Read(byte[], ...)` rather than the other way around... you can see examples of this in the new `Stream.Read/Write` span-based methods in coreclr.

Same issue applies other base virtuals being added.